SceneRefAttributeValidator.cs

```diff
@@ -319,44 +354,92 @@ object existingValue
                 iSerializable?.Clear();
 #if UNITY_EDITOR
                 if (existingValue != null)
-                    EditorUtility.SetDirty(c);
+                {
+                    EditorUtility.SetDirty(component);
+                }
 #endif
                 return null;
             }
 
-            if (iSerializable != null)
+            if (iSerializable == null)
             {
-                if (!iSerializable.OnSerialize(value))
+                bool valuesAreEqual = existingValue != null && (isArray ? ((IEnumerable)value).HaveSameCount(existingValueEnumerable) : value.Equals(existingValue));
+                if (valuesAreEqual)
+                {
                     return existingValue;
+                }
+
+                if (fieldType.IsArray)
+                {
```

Comment:
Minor thing but since this first `if` is the same as the `else` case, I'd modify this to 

```
if (!fieldType.IsArray && typeof(IEnumerable).IsAssignableFrom(fieldType))
{ 
  ...
}
else
{
   field.SetValue(component, value);
}
```